### ISSUE TITLE
OpenNext Ignores force-cache Option on fetch Requests (Data Cache Behavior Differs from Vercel)

### ISSUE THREAD

**Issue by mlukasik-dev** [sst/open-next#493]
I recently migrated my project to OpenNext, deploying it with `sst@v3`. It's a news website built using a custom CMS developed with Next.js. Previously, when hosted on Vercel, nearly all fetch requests to the configuration endpoint were served from cache (with a 99.99% cache hit rate, according to Vercel and our server logs).

However, after the migration, the server is being hit hundreds of times per hour, suggesting that the cache is being bypassed—likely whenever new pages are rendered. This behavior differs significantly from our previous experience on Vercel, where caching worked as expected.

### Problem

- **Expected Behavior**: Fetch requests should be cached, with minimal server hits (as seen on Vercel).
- **Current Behavior**: The server is hit far more frequently, implying the cache is not functioning as intended.

### Questions

1. What could be causing this cache issue after migrating to OpenNext?
2. How can I debug or resolve this problem?

I'd greatly appreciate any help, insights, or advice on this issue.

Thank you in advance!

### Useful Links

- [Next.js Fetch API Documentation](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching#fetch-api)
- [Next.js Fetch Options (cache)](https://nextjs.org/docs/app/api-reference/functions/fetch#optionscache)

**Comment by mlukasik-dev** [comment]
Right, I'm using 3.0.8. I'll try to downgrade. Thank you very much for the reply.

**Comment by conico974** [comment]
@mlukasik-dev You don't have a tag associated with this fetch call right ?
If that's the case the better fix is to just add a tag, this should fix the issue without having to downgrade

**Comment by mlukasik-dev** [comment]
> @mlukasik-dev You don't have a tag associated with this fetch call right ? If that's the case the better fix is to just add a tag, this should fix the issue without having to downgrade

@conico974,  I actually have one or two tags on each fetch. I've checked the DynamoDB and they are written correctly.

**Comment by conico974** [comment]
I see, in this case i think downgrading will do nothing for your issue.
The problem is coming from here https://github.com/sst/open-next/blob/c91c83b5c84c1ea582c03eb23250239954343b64/packages/open-next/src/adapters/cache.ts#L165-L170 We should also add `revalidatedAt` to 1 otherwise every new page make the fetch data cache stale.

I'll make a PR later today for this

**Comment by mlukasik-dev** [comment]
@conico974, Thanks for finding a solution so quickly. I didn't even manage to test a downgrade :)
Unbelievable!

Btw, while browsing the DynamoDB table, I found another bug with cache tags that makes [listing paths by tags](https://open-next.js.org/inner_workings/caching#cloudfront-cache-invalidation) unstable and unpredictable.
I'll create another issue and link it here.

**Comment by mlukasik-dev** [comment]
#495

**Comment by mlukasik-dev** [comment]
@conico974, unfortunately the update didn't fix the problem.

In SST I have set `openNextVersion` to `3.1.1`, new DDB lines do have `revalidatedAt` set to `1`.

But the problem persists. I am getting hundreds of requests to the same configuration endpoint from my website.

**Comment by conico974** [comment]
@mlukasik-dev You'll need to provide a reproduction then

**Comment by conico974** [comment]
If you want to debug in your app you should deploy like that `OPEN_NEXT_DEBUG=true sst deploy`.
This will provide a ton of log and might help identify what's wrong

**Comment by mlukasik-dev** [comment]
@conico974, I've tried to create a very simple [Reproduction Example](https://github.com/mlukasik-dev/open-next-data-cache).
However, everything works as expected so far.

So I thought: maybe the problem is with the previous cache inside DynamoDB? Shouldn't they be removed or something?

**Comment by conico974** [comment]
@mlukasik-dev Here everything is SSG, are you using ISR in your app ?

**Comment by mlukasik-dev** [comment]
@conico974, then I'm not sure I understand what an ISR means...
I thought it was the SSG with pages generated at run time instead of build time (basically my example).

**Comment by mlukasik-dev** [comment]
For my example run time generation is the case because I don't provide `SUPPORTED_LANGS` array to `generateStaticParams` in `app/[lang]/layout.tsx`.

https://github.com/mlukasik-dev/open-next-data-cache/blob/7d51f247f74509ecf3a2a3fd4f81d63118fddcd4/apps/web/src/app/%5Blang%5D/layout.tsx#L24-L26

**Comment by conico974** [comment]
ISR is when you use `revalidate` either in fetch or with `export const revalidate`

**Comment by mlukasik-dev** [comment]
I do use `revalidate` in one place, but not for the problematic (/config) endpoint.

**Comment by conico974** [comment]
Not on this specific fetch, but does the `revalidate` happens inside a page where this config fetch is done ?

**Comment by mlukasik-dev** [comment]
Yes, that page has another fetch with `options.revalidate` set to `86400`.

**Comment by mlukasik-dev** [comment]
I have updated Reproduction Example so it has a fetch with `revalidate` now:
https://github.com/mlukasik-dev/open-next-data-cache/commit/dffa951039cd2d7e4b42e31ff9d23faba1fa3e98

**Comment by mlukasik-dev** [comment]
I checked the ConfigLambda function logs and confirm the bug is in the example.

**Comment by sommeeeer** [comment]
I deployed it too. Only got 2 entries in the `ConfigLambda` log after navigating around to all pages.
